### DISCUSSION TITLE
另辟蹊径部分解决打包问题

### DISCUSSION THREAD

**Discussion by Z-fly** [kanasimi/work_crawler#578] [General]
## 在wsl下打包
前提：安装了WSLg (通过命令行wsl --install安装)
![image](https://user-images.githubusercontent.com/10470892/184499704-349534c4-d4c5-4162-9450-2d460357f92f.png)
安装依赖
`apt install libglib2.0-0 libnss3 libatk1.0-0 libatk-bridge2.0-0 libcups2 libdrm2 libgtk-3-0 libgbm1 libasound2 fonts-wqy-zenhei npm`
进入到目录
```
node work_crawler.updater.js
npm i electron-packager -g
electron-packager .
```
打包完如图
![image](https://user-images.githubusercontent.com/10470892/184499677-5bb56f69-d817-4c30-8187-18b0cafa46cf.png)
使用upx压缩主程序
`upx -9 work_crawler`
![image](https://user-images.githubusercontent.com/10470892/184499791-304216c6-09cf-4740-a30d-22a05db9ce13.png)

## 快捷运行方式
`CreateObject("Wscript.Shell").run "wsl ./work_crawler",vbhide`
![image](https://user-images.githubusercontent.com/10470892/184499888-72323617-0aee-4a4f-bc7c-31a050771658.png)
![image](https://user-images.githubusercontent.com/10470892/184500577-3f34ba55-d109-4014-b658-9acbba4e5246.png)

## 问题
无法识别debian的p7zip(7zr)？
![image](https://user-images.githubusercontent.com/10470892/184500484-6aad2fd1-95ff-4a88-ace4-037ebb517b89.png)
点击打开下载目录没有反，考虑到是wsl下，考虑直接调用windows的explorer.exe
需要apt install npm来解决依赖缺失的问题，如何排除掉不需要的依赖

## 下载地址
 https://cowtransfer.com/s/7a67c5011f3c45 点击链接查看 [ work_crawler-linux-x64.7z ] ，或访问奶牛快传 cowtransfer.com 输入传输口令 x8m9yh 查看；
可以参照上面自行打包，也可以用github的Actions打包。

**Comment by kanasimi** [comment]
感謝您的意見 😄 
這邊已經開啟討論區，您可以貼過去之後關閉這個議題。
由於虛擬機的問題，這邊沒採用WSL。不過歡迎您能一同參與開發！

Archive_file 是在 https://github.com/kanasimi/CeJS/blob/master/application/storage/archive.js#L113

**Reply by Z-fly** [reply]
我发现原来是apt install p7zip-full
顺便你可以在Releases里放electron-packager的包，至少它打出来可以用

**Reply by Z-fly** [reply]
我不太会前端呢，我只能不断尝试

**Reply by kanasimi** [reply]
您可以先試試看，若有覺得需要改進的可以馬上回報。 😄

**Reply by Z-fly** [reply]
> 您可以先試試看，若有覺得需要改進的可以馬上回報。 😄

我是建议把2.14.0的包全部撤掉，因为没有能用的
![image](https://user-images.githubusercontent.com/10470892/185634137-f8cae037-b2f0-4413-b548-056de85d857a.png)

**Comment by kanasimi** [comment]
已轉換到討論區。
